packages/alfa-rules/src/common/predicate/has-namespace.ts

```diff
@@ -1,8 +1,25 @@
 import { Element, Namespace } from "@siteimprove/alfa-dom";
 import { Predicate } from "@siteimprove/alfa-predicate";
+import equals = Predicate.equals;
 
 export function hasNamespace(
-  predicate: Predicate<Namespace> = () => true
-): Predicate<Element> {
+  predicate: Predicate<Namespace>
+): Predicate<Element>;
+export function hasNamespace(
+  ...namespaces: Array<Namespace>
+): Predicate<Element>;
+export function hasNamespace(): Predicate<Element> {
+  let predicate: Predicate<Namespace>;
+
+  if (arguments.length === 0) {
+    predicate = () => true;
+  }
+  if (arguments.length === 1 && typeof arguments[0] === "function") {
+    predicate = arguments[0];
+  }
+  if (arguments.length !== 0 && typeof arguments[0] !== "function") {
+    predicate = equals(...arguments);
+  }
```

Comment:
```suggestion
  predicate: Predicate<Namespace>
): Predicate<Element>;

export function hasNamespace(
  namespace: Namespace,
  ...rest: Array<Namespace>
): Predicate<Element>;

export function hasNamespace(
  namespaceOrPredicate: Namespace | Predicate<Namespace>,
  ...namespaces: Array<Namespace>
): Predicate<Element> {
  let predicate: Predicate<Namespace>;
  
  if (typeof namespaceOrPredicate === "function") {
    predicate = namespaceOrPredicate;
  } else {
    predicate = equals(namespaceOrPredicate, ...namespaces);
  }
```

Reply:
Yes, feels much less hacky (and forces at least one namespace…)
This does however prevent the use of a default predicate. But we don't seem to use it, so I guess it's OK.

Reply:
If we need it, it's easy to introduce an additional overload and a branch that checks for `namespaceOrPredicate === undefined` 👍